### PR TITLE
security.hidepid: enable by default

### PR DESCRIPTION
**Pull request by groxxda**
###### Motivation for this change

The description states that this feature is recommended if there is no valid reason to disable it, so our default should follow this recommendation.
Since improved security is one major achievement of 16.09 I propose to add this to 16.09.

If there are objections against making this the default, we should adjust the description.
If there are only objections against getting this into 16.09 I can rewrite to 17.03.
###### Things done
- [x] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
  - [x] NixOS
  - [ ] OS X
  - [ ] Linux
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

CC @joachifm @edolstra

### CODE REVIEW

**Comment by joachifm** [comment]
To be clear, the "recommendation" in the description is my *opinion*, so I'd take it with a grain of salt ... Anyway, I'm obviously in favour of this change.  I set the default to false primarily for strategic reasons. I've used this feature for years without ever experiencing any problems.

File: nixos/modules/security/hidepid.nix [hunk 1]
```diff
@@ -3,18 +3,23 @@ with lib;
...
-    security.hideProcessInformation = mkEnableOption "" // { description = ''
-      Restrict access to process information to the owning user.  Enabling
-      this option implies, among other things, that command-line arguments
-      remain private.  This option is recommended for most systems, unless
-      there's a legitimate reason for allowing unprivileged users to inspect
-      the process information of other users.
+    security.hideProcessInformation = mkOption {
+      default = true;
+      example = false;
+      description = ''
```
Comment (joachifm) [inline nixos/modules/security/hidepid.nix]: Ah, this reminds me I should have moved this description to separate module documentation. I sometimes forget that the option description field isn't a blog :)

**Comment by groxxda** [comment]
I know this is a policy decision, that's why I mentioned eelco.
My thought was: If this is a recommendation in the manual, we should go with it. 
Should I take this to the mailing list?

**Comment by dezgeg** [comment]
I'm not a fan of changes that make us diverge from upstream defaults or what other (major) distros do, even the minor things tend to pile up in the end and add a lot of cognitive load for those that have to deal with several kinds of systems each day.

**Comment by groxxda** [comment]
since we won't reach consensus i'm closing this.

**Pull request closed without merging**